File: sdk/python/compliagl/clients/authorizations.py
from __future__ import annotations
from typing import Any
from .base import BaseClient


class AuthorizationClient(BaseClient):
    path = "execution-authorizations"

    def create(self, data: dict[str, Any], *, idempotency_key: str | None = None):
        return self._post(self.path, data, idempotency_key=idempotency_key)

    def issue(self, data: dict[str, Any], *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/issue", data, idempotency_key=idempotency_key)

    def list(self, *, skip: int | None = None, limit: int | None = None):
        return self._get(self.path, {"skip": skip, "limit": limit})

    def get(self, id: str):
        return self._get(f"{self.path}/{id}")

    def verify(self, id: str, expected_fields: dict[str, Any] | None = None, activate: bool = False, *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/{id}/verify", {"expected_fields": expected_fields or {}, "activate": activate}, idempotency_key=idempotency_key)

    def consume(self, id: str, *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/{id}/consume", {}, idempotency_key=idempotency_key)

    def revoke(self, id: str, reason: str | None = None, *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/{id}/revoke", {"reason": reason}, idempotency_key=idempotency_key)

    def transition(self, id: str, status: str, *, idempotency_key: str | None = None):
        return self._post(f"{self.path}/{id}/transition", {"status": status}, idempotency_key=idempotency_key)

File: sdk/typescript/src/clients/authorizations.ts
import { CrudClient } from './base';
import type { ExecutionAuthorization, IssueExecutionAuthorization } from '../models';
import type { RequestOptions } from '../config';
export class AuthorizationClient extends CrudClient<ExecutionAuthorization, IssueExecutionAuthorization> {
  constructor(http: import('../http').HttpClient) { super(http, '/execution-authorizations'); }
  issue(input: IssueExecutionAuthorization, options?: RequestOptions): Promise<ExecutionAuthorization> { return this.http.post<ExecutionAuthorization>('/execution-authorizations/issue', input, options); }
  verify(id: string, options?: RequestOptions): Promise<{ valid: boolean; authorization: ExecutionAuthorization }> { return this.http.post<{ valid: boolean; authorization: ExecutionAuthorization }>('/execution-authorizations/verify', { id }, options); }
  consume(id: string, options?: RequestOptions): Promise<ExecutionAuthorization> { return this.http.post<ExecutionAuthorization>('/execution-authorizations/consume', { id }, options); }
  revoke(id: string, reason?: string, options?: RequestOptions): Promise<ExecutionAuthorization> { return this.http.post<ExecutionAuthorization>('/execution-authorizations/revoke', { id, reason }, options); }
  transition(id: string, status: string, options?: RequestOptions): Promise<ExecutionAuthorization> { return this.http.post<ExecutionAuthorization>('/execution-authorizations/transition', { id, status }, options); }
}
